doc/examples/applications/plot_3d_image_processing.py

```diff
@@ -0,0 +1,311 @@
+
+#####################################################################
+# `Gamma correction <https://en.wikipedia.org/wiki/Gamma_correction>`_
+# brightens or darkens an image. A power-law transform, where `gamma` denotes
+# the power-law exponent, is applied to each pixel in the image: `gamma < 1`
+# will brighten an image, while `gamma > 1` will darken an image.
+
+
+def plot_hist(ax, data, title=None):
+    # Helper function for plotting histograms
+    ax.hist(data.ravel(), bins=256)
+    ax.ticklabel_format(axis="y", style="scientific", scilimits=(0, 0))
+
+    if title:
+        ax.set_title(title)
+
+
+gamma_low_val = 0.5
+gamma_low = exposure.adjust_gamma(data, gamma=gamma_low_val)
+
+gamma_high_val = 1.5
+gamma_high = exposure.adjust_gamma(data, gamma=gamma_high_val)
+
+_, ((a, b, c), (d, e, f)) = plt.subplots(nrows=2, ncols=3, figsize=(12, 8))
+
+show_plane(a, data[32], title="Original")
+show_plane(b, gamma_low[32], title="Gamma = {}".format(gamma_low_val))
+show_plane(c, gamma_high[32], title="Gamma = {}".format(gamma_high_val))
+
+plot_hist(d, data)
+plot_hist(e, gamma_low)
+plot_hist(f, gamma_high)
+# sphinx_gallery_thumbnail_number = 4
+
+#####################################################################
+# `Histogram
+# equalization <https://en.wikipedia.org/wiki/Histogram_equalization>`_
+# improves contrast in an image by redistributing pixel intensities. The most
+# common pixel intensities get spread out, letting areas of lower local
+# contrast gain a higher contrast. This may enhance background noise though.
+
+equalized_data = exposure.equalize_hist(data)
+
+display(equalized_data)
+
+#####################################################################
+# As before, if we have a Jupyter kernel running, we can explore the above
+# slices interactively.
+
+explore_slices(equalized_data);
+
+#####################################################################
+# Let us now plot the image histogram before and after histogram equalization.
+# Below, we plot the respective cumulative distribution functions (CDF).
+
+_, ((a, b), (c, d)) = plt.subplots(nrows=2, ncols=2, figsize=(16, 8))
+
+plot_hist(a, data, title="Original histogram")
+plot_hist(b, equalized_data, title="Equalized histogram")
+
+cdf, bins = exposure.cumulative_distribution(data.ravel())
+c.plot(bins, cdf, "r")
+c.set_title("Original CDF")
+
+cdf, bins = exposure.cumulative_distribution(equalized_data.ravel())
+d.plot(bins, cdf, "r")
+d.set_title("Histogram equalization CDF")
+
+#####################################################################
+# Most experimental images are affected by salt and pepper noise. A few bright
+# artifacts can decrease the relative intensity of the pixels of interest. A
+# simple way to improve contrast is to clip the pixel values on the lowest and
+# highest extremes. Clipping the darkest and brightest 0.5% of pixels will
+# increase the overall contrast of the image.
+
+vmin, vmax = np.percentile(data, q=(0.5, 99.5))
```

Comment:
Personally, I prefer using quantiles, though it's a personal preference, consider this comment optional. =)

```suggestion
vmin, vmax = np.quantile(data, q=(0.005, 0.995))
```

Reply:
I went for a popularity contest to decide and percentile won. ;-)
I searched for `"cell biology" "percentile"` vs `"cell biology" "quantile"`.

Reply:
I also did not know that numpy had a quantile function :-). (but I'm happy that I learned this, since I also prefer a bit quantile to percentile, but I'm fine with letting the pop contest decide.)